src/tests/testRefCount.cc

```diff
@@ -55,7 +55,7 @@ testRefCount::testCountability()
         ToRefCount anObject(new _ToRefCount);
         CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
         CPPUNIT_ASSERT_EQUAL(1, anObject->someMethod());
-        anObject = anObject;
+        anObject = *&anObject;  //Casting is to prevent -Wself-assign-overloaded warnings
```

Comment:
This is not really "casting". Let's clarify (and improve formatting):

```suggestion
        anObject = *&anObject;  // test self-assign without -Wself-assign-overloaded warnings
```